experimental/JuLie/partitions.jl

```diff
@@ -0,0 +1,669 @@
+  length(mu)==length(v) || throw(ArgumentError("mu and v should have the same length"))
+  m>=0 || throw(ArgumentError("m ≥ 0 required"))
+  n>=1 || throw(ArgumentError("n ≥ 1 required"))
+
+  T = typeof(m)
+  if isempty(mu)
+    return Partition{T}[]
+  end
+
+  r = length(v)
+  j = 1
+  k = mu[1]
+  ll = v[1]
+  x = zeros(Int8, n+1)
+  y = zeros(Int8, n+1)
+  ii = zeros(Int8, n)
+  i_1 = 0
+  P = Partition{T}[]
+
+  num = 0
+  gotob2 = false
+  gotob1 = true
+
+  for i = n:-1:1
+    x[i] = ll
+    y[i] = ll
+    k = k - 1
+    m = m - ll
+    if k == 0
+      if j == r
+        return P
+      end
+      j = j + 1
+      k = mu[j]
+      ll = v[j]
+    end
+  end #for i
+
+  lr = v[r]
+  ll = v[1]
+
+  if m < 0 || m > n * (lr - ll)
+    return P
+  end
+
+  if m == 0
+    push!(P,Partition{T}(x[1:n]))
+    return P
+  end
+
+  i = 1
+  m = m + y[1]
+
+  while gotob1 == true
+    if !gotob2
+      for j = mu[r]:-1:1
+        if m<=lr
+          gotob2 = true
+          break
+        end
+        x[i] = lr
+        ii[i] = r - 1
+        i = i + 1
+        m = m - lr + y[i]
+      end #for j
+
+      if !gotob2
+        r = r - 1
+      end
+
+      gotob2 = true
+    end #if
+
+    if gotob2
+      while r>0 && v[r] > m
+        r = r - 1
+      end
+
+      if r == 0
+        break
+      end
+
+      lr = v[r]
+      if m == lr
+        x[i] = lr
+        if i<=n
+          push!(P, Partition{T}(x[1:n]))
+        else
+          break
+        end
+
+        r = r - 1
+        if r==0
+          break
+        end
+        lr = v[r]
+      end #if
+
+      k = y[i]
+      if lr > k && m - lr <= (n-i)*(lr - ll)
+        gotob2 = false
+        continue
+      else
+        x[i] = k
+      end #if
+      i_1 = i - 1
+      for i_0 = i_1:-1:1
+        i = i_0
+        r = ii[i]
+        lr = v[r]
+        m = m + x[i] - k
+        k = y[i]
+        if lr > k && m - lr <= (n-i)*(lr-ll)
+          gotob2 = false
+          break
+        else
+          x[i] = k
+        end #if
+      end #for
+      if gotob2
+        gotob1 = false
+      end
+    end #if gotob2
+  end #while
+  return P
+end
+
+
+
+@Markdown.doc """
+    dominates(lambda::Partition, mu::Partition)
+
+The **dominance order** on partitions is the partial order ``⊵`` defined by ``λ ⊵ μ`` if and only if ``λ₁ + … + λᵢ ≥ μ₁ + … + μᵢ`` for all i. This function returns true if ``λ ⊵ μ``.
+
+# References
+1. Wikipedia, [Dominance order](https://en.wikipedia.org/wiki/Dominance_order)
+"""
+function dominates(lambda::Partition, mu::Partition)
+  dif = 0
+  i = 1
+  while i <= min(length(lambda), length(mu))
+    dif += lambda[i] - mu[i]
+    i += 1
+    if dif < 0
+      return false
+    end
+  end
+  if length(lambda) < length(mu)
+    while i <= length(mu)
+      dif -= mu[i]
+      i += 1
+    end
+    if dif < 0
+      return false
+    end
+  end
+  return true
+end
+
+
+@Markdown.doc """
+    conjugate(lambda::Partition{T}) where T<:IntegerUnion
```

Comment:
The following is a purely informal remark, I don't expect you to do anything about it:

Note that in Julia (and OSCAR) this is normally called `conj`. However, I've added a suggestion to #1826 that we should allow `conjugate` as a synonym for `conj`.